### ISSUE TITLE
Can't connect to servers that require TLSv1+

### ISSUE THREAD

**Issue by jprenken** [xchataqua/xchataqua#152]
It looks like X-Chat Aqua/Azure only supports SSLv3; trying to negotiate a connection with a server that requires TLSv1 or newer (with SSL_OP_NO_SSLv3 set on the server side) fails:

Connection failed. Error: (336151568) error:14094410:SSL routines:SSL3_READ_BYTES:sslv3 alert handshake failure

**Comment by kuoirad** [comment]
But will it ever be fixed here?  Hexchat is kinda fugly compared to Azure.  SSL in general for Azure seems to be a bit of a pain.  Is it still an active project?

**Comment by TingPing** [comment]
@kuoirad You misunderstand, that branch is Azure with the core of hexchat which is updated to solve this problem.

**Comment by kuoirad** [comment]
Ahhh....my apologies.  I will see if I can go snag a binary of that then.

**Comment by kuoirad** [comment]
@TingPing - forgive what's probably a stupid question, but I haven't been able to find a binary - does one exist anywhere, or do I have to compile my own?  If I'm compiling, are the instructions at https://github.com/xchataqua/xchataqua/wiki/HowToBuild still valid?

**Comment by youknowone** [comment]
It is valid in master branch -  I hope. If you can't do it, please let me know whats wrong.
It is not vaild in hexchat branch yet.

**Comment by kuoirad** [comment]
I'll try and set aside a little time this afternoon to test it out for you on the master branch - don't think I have XCode installed on this machine anymore.

**Comment by Xoib** [comment]
Hi, @jprenken, @TingPing, @kuoirad, @youknowone: Does the state of this issue involved ?
Because for what I can see, the main latest release, the head of master and hexchat are still not patched.

**Comment by TingPing** [comment]
@Xoib Considering this issue is still opened it is safe to assume that it has not be fixed.

**Comment by Xoib** [comment]
Of course. However your first and second comments seems to indicate that it is already fixed.
I agree that hexchat core is patched, but as the master doesn't use hexchat core and because release channel is built upon master we should make the level of severity of this issue far more higher.

**Comment by LMariachi** [comment]
In the meantime, is there any way to disable SSLv3 connection attempts without unchecking SSL altogether?

**Comment by dlangille** [comment]
Any ETA on release for the xchat branch?  In order to use TLS, I have had to start using something else.

**Comment by BigValen** [comment]
Newer versions of znc don't support SSLv3 anymore, so expect to see more people noticing this...

**Comment by dlangille** [comment]
FWIW, I've started using Textual.

**Comment by Armada651** [comment]
It should be possible to fix this issue using the same patch used in the Ubuntu repository, please see the above PR.

**Comment by Armada651** [comment]
I just got around to testing it, using the above patch I can connect to my ZNC bouncer again without having to build the hexchat branch.

**Comment by rcombs** [comment]
Issue affects me here as well.

**Comment by dlangille** [comment]
I am unsubscribing not because the issue is fixed, but because I'm using a different IRC client now.  The issue persists and is important to many, but I need to move on.

**Comment by Armada651** [comment]
Perhaps this project should be forked? This issue can easily be fixed, but I don't think it's being maintained anymore.

**Comment by JohnnyricoMC** [comment]
POODLE has been a publicly disclosed vulnerability since Oct. 14 2014, this has been fixed in the Hexchat master branch for months now. Any server administrator worth his salt disabled SSLv3 on his servers months ago.

The Mac App Store version was last updated on Mar 16 2015 (https://itunes.apple.com/us/app/xchat-azure/id447521961) , yet the last commit on Github dates back to May 15, 2014. If it is still being maintained somehow, I don't think it's happening through Github.

**Comment by youknowone** [comment]
Changing core to hexchat will fix it probably. It was under working in hexchat branch: https://github.com/xchataqua/xchataqua/tree/hexchat

due to releasing issue.

Any patch or even fork will be appriciated (unless I can merge it). I may can't finish it soon.

**Comment by Armada651** [comment]
@youknowone Please have a look at my patch: https://github.com/xchataqua/xchat/pull/1

I have already confirmed the problem is fixed with those simple changes.

**Comment by youknowone** [comment]
Oh.. I am sorry to miss that one :/ I only checked this repository, never xchat one.

**Comment by Armada651** [comment]
No problem, thank you for merging it, now all that's needed is a new release with the patched xchat core.

**Comment by youknowone** [comment]
I will release this patch with El Capitan plugin support update.

**Comment by kuoirad** [comment]
Any chance we can get this release somewhere in the near future?  Or are we literally waiting for El Capitan's release, which is now slated for the end of September?

**Comment by Armada651** [comment]
I don't think that's too bad, we're already very close to the release of El Captian. Releasing a new version of an app takes some time, so it would be better to combine it with the El Captian release.

**Comment by youknowone** [comment]
Apple is getting El capitan submitting of new versions now :)

**Comment by youknowone** [comment]
1.17.0 is waiting for review: https://github.com/xchataqua/xchataqua/releases/tag/appstore-1.17.0
